### PR TITLE
modal: change `data-dismiss` so that it can be outside of a modal using `bs-target`

### LINKED ISSUE #33324: Modal data-bs-dismiss and data-bs-target
Currently modals support a 'data-bs-target' with the 'data-bs-toggle' attribute, that way if you have a 'data-bs-toggle' of 'Modal' it knows which Modal to actually open.

I would love to see 'data-bs-dismiss' also support a 'data-bs-target' attribute if one is present.

Right now, I am free to open a modal, then while that modal is being displayed, open another one and have the new one on top, which is great for at work, in my case, modalOne is for adding a not on file part number to a customer quote, and modalTwo is for a markup calculator that modalOne can open (This is for our stores), I used to do all this using Bootbox, but since moving to BS5 for all my new stuff and completely dropping jQuery, I can't use Bootbox anymore.

The problem is that if you close modalTwo, it also closes modalOne, so what I have done is when modalTwo opens, I add a custom event to the btn-close (I removed 'data-bs-dismiss' from the button) and also overrode the 'escape' key, in both instances when you click the 'X' on modalTwo or press escape, it closes only modalTwo and takes you back to the correct field on modalOne.

If there was a 'data-bs-target' available for 'data-bs-dismiss', then I could tell it which modal it is to close.

Hopefully that all makes sense.

### PR DESCRIPTION
**Pull request by GeoSot**
Closes #33324

### CODE REVIEW

**Comment by GeoSot** [comment]
@twbs/js-review do we want this?

**Comment by rohit2sharma95** [comment]
Before it was not possible to place the backdrop inside any specific HTML element, so the dismiss button was always in the same modal (that does not require the `data-bs-target` on the dismiss button). But now the button can be outside the modal as well so it makes sense to add a target element on the dismiss button IMO.

/CC @XhmikosR

File: js/src/modal.js [hunk 1]
```diff
@@ -434,6 +433,13 @@ EventHandler.on(document, EVENT_CLICK_DATA_API, SELECTOR_DATA_TOGGLE, function (
   data.toggle(this)
 })
 
+EventHandler.on(document, EVENT_CLICK_DATA_API, SELECTOR_DATA_DISMISS, function (event) {
+  const target = getElementFromSelector(this) || this.closest(SELECTOR)
+  const modal = Modal.getInstance(target) || new Modal(target)
```
Comment (alpadev) [inline js/src/modal.js:438]: Why not `getOrCreateInstance`? 😁

Comment (GeoSot) [reply js/src/modal.js:438]: hahahah, cause the pr is older than `getOrCreateInstance`

I 'll update it ;)

**Comment by ffoodd** [comment]
Chaining modals is usually a bad idea. The PR looks great but I'm more concerned about what it allows: the use-case mentioned in the related issues feels very weird to me. @patrickhlauke Do you have any insight on  something like a button in a modal that'd open another modal, keeping the first one opened above?

**Comment by GeoSot** [comment]
Just keep in mind that alert is already working this way. 

It tries to parse data-target and if it is not exist, tires to get the closest `.alert` element

**Comment by ffoodd** [comment]
Yeah but alerts aren't modals: it does not require everything apart itself to be inactive, whereas modal do.

**Review by ffoodd — APPROVED** [review]
@GeoSot Does this need documentation, on the tech side? Regarding accessibility, I think we're safe as long as we do not promote exotic use cases and make this obvious in our docs.

**Comment by GeoSot** [comment]
@ffoodd I 've added some documentation to support this 'feature'

File: site/content/docs/5.0/components/modal.md [hunk 2]
```diff
@@ -831,11 +831,27 @@ The modal plugin toggles your hidden content on demand, via data attributes or J
...
+```html
+<button type="button" class="btn-close" data-bs-dismiss="modal" aria-label="Close"></button>
+```
+
+or on a button **outside the modal** using the `data-bs-target` as demonstrated below:
+
+```html
+<button type="button" class="btn-close" data-bs-dismiss="modal" data-bs-target="#my-modal" aria-label="Close"></button>
+```
+We provide both ways, handling dismissal inside and outside modal, although suggesting avoid using the second, as it may lead you to accessibility issues
```
Comment (ffoodd) [inline site/content/docs/5.0/components/modal.md:854]:
```suggestion
{{< callout warning >}}
While both ways to dismiss a modal are supported, keep in mind that dismissing from outside a modal does not match [the WAI-ARIA modal dialog design pattern](https://www.w3.org/TR/wai-aria-practices-1.1/#dialog_modal). Do this at your own risk.
{{< /callout >}}
```

**Review by ffoodd — APPROVED** [review]
I suggested a few changes regarding headings hierarchy and callout :+1: 

Maybe there's more to improve regarding docs but it sounds good to me.

**Pull request merged**